### PR TITLE
SFT additions to the User Guide

### PR DESCRIPTION
**Pull request by DellaBitta**
[ Description ]
Updating the User Guide to reference the new SFT features.

[ JIRA ]
XCL-3063
[ SFT ] User Guide Updates

[ Reviewer ]
@atigyi 

[ QA ]
None.

[ Release Notes ]
User Guide Updates for Secure File Transfer Features.

### CODE REVIEW

File: doc/user_guide.md [hunk 1]
```diff
@@ -6,24 +6,24 @@
...
 
-The Xively Client Agent is a C library that provides MQTT connectivity over TLS as a baseline, but it includes much more on top of the standard MQTT Client:
+The Xively Client is a C library which provides MQTT connectivity over TLS.  Compared to other MQTT cilents, it brings these other features to the IoT table:
 
 ### Flexibility
 
 The client can scale capabilities to meet the needs of the platform
 
-* It can operate on a single thread and our implementation runs in non-blocking mode by default.
+* It can operate on a single thread and also runs in non-blocking mode by default.
```
Comment (atigyi) [inline doc/user_guide.md]: 👍

File: doc/user_guide.md [hunk 2]
```diff
@@ -33,6 +33,16 @@ Through the use of [coroutines](http://en.wikipedia.org/wiki/Coroutine) this MQT
 * The library can send and receive simultaneously on a single socket.
 * Communication requirements of your application will not interfere with the usability of your device.
 
+### Secure File Transfer (SFT)
```
Comment (atigyi) [inline doc/user_guide.md]: I don't know, just a suggestion: if one searches for Firmware Update then he hardly finds this "flagship" feature. So I'd mention 'Firmware Update (FWU)' expression too somewhere. Maybe in this section title, why isn't it:
'### Secure File Transfer (SFT), Firmware Update (FWU)' ?
Just an idea.

File: doc/user_guide.md [hunk 3]
```diff
@@ -33,6 +33,16 @@ Through the use of [coroutines](http://en.wikipedia.org/wiki/Coroutine) this MQT
...
+
+The Xively C Client comes pre-built with functionality to communicate with the [Xively SFT Service](https://developer.xively.com/v1.0/docs/how-to-securely-transfer-files-to-my-devices).  
+
+* Devices can be easily updated in the field with new firmware and support files.
+* Data is downloaded in the background, no need to create a second connection to a new service.
+* Download and update status is reported by the Xively C Client to the SFT service automatically, helping you to monitor the progress of a software deployment across your fleet of devices.
```
Comment (atigyi) [inline doc/user_guide.md]: well phrased as usual 👍

File: doc/user_guide.md [hunk 4]
```diff
@@ -50,7 +60,9 @@ The Xively Client was written with the understanding that the IoT Field might ch
...
+Currently the optimized client storage footprint requirements are about 30kb for embedded devices with optimized toolchains, not including TLS functionality.  This footprint includes a TLS certificate for validating the Xively server during TLS, 3 backup certificates for use if the main certificate becomes compromised, an event dispatcher and scheduler, the connection backoff system, and both Platform networking, time, random number generator and memory implementations.  This footprint does not include a TLS implementation, but if one exists on the platform already, then the size requirements are negligible.
```
Comment (atigyi) [inline doc/user_guide.md]: "...and both Platform networking..." 'both' ? There is an long enumeration afterwards :) Sorry I know you didn't alter this part, but I found it sorry.

File: doc/user_guide.md [hunk 5]
```diff
@@ -50,7 +60,9 @@ The Xively Client was written with the understanding that the IoT Field might ch
...
+
+The Secure File Transfer
```
Comment (atigyi) [inline doc/user_guide.md]: hello, who are you dear lonely "The Secure File Transfer"?! :D

File: doc/user_guide.md [hunk 6]
```diff
@@ -142,6 +154,21 @@ With this communication toolbox embedded devices can be used to listen and publi
...
+
+The Xively C Client uses the device's existing MQTT connection for SFT functionality, so there's no need to build a separate connection to another service.  And since the Xively C Client is non-blocking and asynchronous, the files are downloaded in the background without needing to change any application flow or interrupt device usability.  
+
+In Xively SFT, the client drives the whole process, determining when to download files, how large of a chunk to fetch at a time, and when to reboot to test new firwmare.  Download status and errors are automatically sent by the Xively C Client so that you can track the software deployment progress across your whole fleet via the Xively SFT Service interface.
+
+We have chosen some suggested configurations for a default behavior, but the code can be easily customized for the best-fit of your platform and product.
```
Comment (atigyi) [inline doc/user_guide.md]: 👍

File: doc/user_guide.md [hunk 7]
```diff
@@ -142,6 +154,21 @@ With this communication toolbox embedded devices can be used to listen and publi
...
+
+In Xively SFT, the client drives the whole process, determining when to download files, how large of a chunk to fetch at a time, and when to reboot to test new firwmare.  Download status and errors are automatically sent by the Xively C Client so that you can track the software deployment progress across your whole fleet via the Xively SFT Service interface.
+
+We have chosen some suggested configurations for a default behavior, but the code can be easily customized for the best-fit of your platform and product. 
+
+The code for the file storage and firwmare update functionality is handled through the File IO and Firmware Update Board Support Package implementations. For more information please see the [Xively C Client Porting Guide](https://github.com/xively/xively-client-c/blob/master/doc/porting_guide.md) and the functions declared in `include/bsp/xi_bsp_io_fs.h` and `include/bsp/xi_bsp_fwu.h`.
```
Comment (atigyi) [inline doc/user_guide.md]: add "(BSP)" after the "Board Support Package"

File: doc/user_guide.md [hunk 8]
```diff
@@ -142,6 +154,21 @@ With this communication toolbox embedded devices can be used to listen and publi
...
+
+In Xively SFT, the client drives the whole process, determining when to download files, how large of a chunk to fetch at a time, and when to reboot to test new firwmare.  Download status and errors are automatically sent by the Xively C Client so that you can track the software deployment progress across your whole fleet via the Xively SFT Service interface.
+
+We have chosen some suggested configurations for a default behavior, but the code can be easily customized for the best-fit of your platform and product. 
+
+The code for the file storage and firwmare update functionality is handled through the File IO and Firmware Update Board Support Package implementations. For more information please see the [Xively C Client Porting Guide](https://github.com/xively/xively-client-c/blob/master/doc/porting_guide.md) and the functions declared in `include/bsp/xi_bsp_io_fs.h` and `include/bsp/xi_bsp_fwu.h`.  
+
+Additionally we have a few reference implementations for POSIX and the TI CC3200 in the `src/bsp/platform/posix/` and the `src/bsp/platform/cc3200` directories, and a [SFT Tutorial on the Xively Developer Center](https://developer.xively.com/v1.0/docs/ti-cc3200-sft-example).
```
Comment (atigyi) [inline doc/user_guide.md]: either with or without trailing slash for directory names. I like both, jut keep them consistent, thx.

File: doc/user_guide.md [hunk 9]
```diff
@@ -142,6 +154,21 @@ With this communication toolbox embedded devices can be used to listen and publi
 
 Currently the Xively Service supports a maximum message size of 128kb.
 
+
+### Secure File Transfer (SFT)
```
Comment (atigyi) [inline doc/user_guide.md]: What about this subsection title?
'### Secure File Transfer (SFT), Firmware Update (FWU)' ?

File: doc/user_guide.md (hunk 3 above)
Comment (DellaBitta) [reply doc/user_guide.md]: Thanks, took me a while to carve this out actually.

File: doc/user_guide.md (hunk 5 above)
Comment (DellaBitta) [reply doc/user_guide.md]: lol  woops.

File: doc/user_guide.md (hunk 8 above)
Comment (DellaBitta) [reply doc/user_guide.md]: cool, thanks.  Yeah I wanted to drive home that they're directories and not files, thanks for finding the inconsistency!

**Pull request merged**